includes/discovery/sensors/state/cisco.inc.php

```diff
@@ -117,6 +122,63 @@
+                ['value' => 3, 'generic' => 0, 'graph' => 0, 'descr' => 'online'],
+                ['value' => 4, 'generic' => 1, 'graph' => 0, 'descr' => 'low power mode'],
+            ];
+        } elseif ($state_name == 'c3gGsmCurrentBand') {
+            $states = [
+                ['value' => 1, 'generic' => 3, 'graph' => 0, 'descr' => 'unknown'],
+                ['value' => 2, 'generic' => 2, 'graph' => 0, 'descr' => 'invalid'],
+                ['value' => 3, 'generic' => 3, 'graph' => 0, 'descr' => 'none'],
+                ['value' => 4, 'generic' => 0, 'graph' => 0, 'descr' => 'gsm850'],
+                ['value' => 5, 'generic' => 0, 'graph' => 0, 'descr' => 'gsm900'],
+                ['value' => 6, 'generic' => 0, 'graph' => 0, 'descr' => 'gsm1800'],
+                ['value' => 7, 'generic' => 0, 'graph' => 0, 'descr' => 'gsm1900'],
+                ['value' => 8, 'generic' => 0, 'graph' => 0, 'descr' => 'wcdma800'],
+                ['value' => 9, 'generic' => 0, 'graph' => 0, 'descr' => 'wcdma850'],
+                ['value' => 10, 'generic' => 0, 'graph' => 0, 'descr' => 'wcdma1900'],
+                ['value' => 11, 'generic' => 0, 'graph' => 0, 'descr' => 'wcdma2100'],
+                ['value' => 12, 'generic' => 0, 'graph' => 0, 'descr' => 'lte band'],
+            ];
+        } elseif ($state_name == 'c3gGsmPacketService') {
+            $states = [
+                ['value' => 1, 'generic' => 3, 'graph' => 0, 'descr' => 'unknown'],
+                ['value' => 2, 'generic' => 1, 'graph' => 0, 'descr' => 'none'],
+                ['value' => 3, 'generic' => 0, 'graph' => 0, 'descr' => 'gprs'],
+                ['value' => 4, 'generic' => 0, 'graph' => 0, 'descr' => 'edge'],
+                ['value' => 5, 'generic' => 0, 'graph' => 0, 'descr' => 'umts wcdma'],
+                ['value' => 6, 'generic' => 0, 'graph' => 0, 'descr' => 'hsdpa'],
+                ['value' => 7, 'generic' => 0, 'graph' => 0, 'descr' => 'hsupa'],
+                ['value' => 8, 'generic' => 0, 'graph' => 0, 'descr' => 'hspa'],
+                ['value' => 9, 'generic' => 0, 'graph' => 0, 'descr' => 'hspa plus'],
+                ['value' => 10, 'generic' => 0, 'graph' => 0, 'descr' => 'lte'],
+            ];
+        } elseif ($state_name == 'c3gGsmCurrentRoamingStatus') {
+            $states = [
+                ['value' => 1, 'generic' => 3, 'graph' => 0, 'descr' => 'unknown'],
+                ['value' => 2, 'generic' => 0, 'graph' => 0, 'descr' => 'roaming'],
+                ['value' => 3, 'generic' => 0, 'graph' => 0, 'descr' => 'home'],
+            ];
+        } elseif ($state_name == 'c3gGsmSimStatus') {
+            $states = [
+                ['value' => 1, 'generic' => 3, 'graph' => 0, 'descr' => 'unknown'],
+                ['value' => 2, 'generic' => 0, 'graph' => 0, 'descr' => 'ok'],
+                ['value' => 3, 'generic' => 0, 'graph' => 0, 'descr' => 'not inserted'],
```

Comment:
Note that this is not right:

.1.3.6.1.4.1.9.9.661.1.3.2.1.6.<idx> =  c3gGsmSimStatus  : refers to :

Current service type: 
unknown(1) - service type is unknown 
invalid(2) - no service 
circuitSwitched(3) - circuit switched service 
packetSwitched(4) - packet switched service 
combined(5) - combination of circuit and packet switched service

source:
http://oidref.com/1.3.6.1.4.1.9.9.661.1.3.2.1.14

I've detected this when testing in a real device:
My LibreNMS shows "SIM Status" = 3 = not inserted .

Router SIM is OK:  telnet info:

Modem Security Information
==========================
Active SIM = 0
SIM switchover attempts = 0
Card Holder Verification (CHV1) = Enabled
SIM Status = OK
SIM User Operation Required = None
Number of CHV1 Retries remaining = 3

Reply:
Hi @Cupidazul, thank you for your feedback and great to see others value the enhanced cellular information. I will take a look at your comment and will look how that oid would turn out in my environment, and yes, I also have these 899's and a bunch of other cellular devices available. What would your suggestion be? Adding the extra status information as a sensor and making the output of the c3gGsmSimStatus better by looking at the status of other variables like IMSI and change "not inserted" to "ok" if there is clearly a sim inserted? My guess is that the faulty reporting is IOS dependent as I have not seen the wrong reporting myself.

Reply:
Hi @Schouwenburg , yes I use these OID's frequently on other systems, It's awesome, someone finally got this working in LibreNMS. I have other two device types I usually collect info, Huawei and TELDAT, actually I have all the information about MIB's , Status, RSSI and other values, your work is a very nice starting point. Congrats to you! Very Nice!

Here are my suggestions:
1. We should create a new discovery/poller/menu type, like 'gsm' or 'celullar', instead of 'wireless', this deserves something better.
2. As we can see, in this case we are talking about SIM status, and we could try to adapt for each device type, in this case the 899's have another information status for the atual OID (yes the router has a SIM sinserted, for sure, as we can see in the telnet output).
3. We should divide the information about GSM, each in their groups, SIM Info , Cell Info , Service Info , Roaming Info, etc...

And, Yeah I know what you mean, unfortunately, I've seen sometimes, from one device to another, and even with different IOS's things might change.

But in this paticular MIB  the actual code states:   .1.3.6.1.4.1.9.9.661.1.3.2.1.6. = c3gGsmSimStatus 
 ['num_oid' => '.1.3.6.1.4.1.9.9.661.1.3.2.1.6.',    'oid' => 'c3gGsmSimStatus',                      'state_name' => 'c3gGsmSimStatus',              'mib' => 'CISCO-WAN-3G-MIB',                'descr' => 'SIM status'],

We might need to double check this, as, I've found this information, that looks to contradict our actual code[:](http://www.circitor.fr/Mibs/Html/C/CISCO-WAN-3G-MIB.php)

c3gGsmCurrentRoamingStatus | 1.3.6.1.4.1.9.9.661.1.3.2.1.6
-- | --
This object indicates whether the modem is in the home network or is roaming.
OBJECT-TYPE |   |   |  
  | INTEGER | unknown(1), roaming(2), home(3)

---

c3gGsmSimStatus | 1.3.6.1.4.1.9.9.661.1.3.5.1.1.2
-- | --
SIM status. Indicates whether the SIM is present or removed from the SIM socket, and its current status.
OBJECT-TYPE |   |   |  
  | INTEGER | unknown(1), ok(2), notInserted(3), removed(4), initFailure(5), generalFailure(6), locked(7), chv1Blocked(8), chv2Blocked(9), chv1Rejected(10), chv2Rejected(11), mepLocked(12), networkRejected(13) |  


Keep Up this nice Work! yup I'm using it ! its awesome.
Cheers!

Reply:
Hi @Schouwenburg , reminder.

Reply:
Corrected in ([#12724](https://github.com/librenms/librenms/pull/12724))